src/Snak/SnakList.php

```diff
@@ -15,17 +19,247 @@
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  * @author Addshore
  */
-class SnakList extends HashArray {
+class SnakList extends ArrayObject implements Hashable, Comparable {
 
 	/**
-	 * @see GenericArrayObject::getObjectType
+	 * Maps snak hashes to their offsets.
+	 *
+	 * @var array [ snak hash (string) => array [ snak offset (string|int) ] | snak offset (string|int) ]
```

Comment:
This can not be an array of arrays any more. Multiple `is_array` checks below are not needed any more. See #712.